backend/layers/processing/schema_migration.py

```diff
@@ -277,32 +285,37 @@ def wrapper(*args, **kwargs):
             except Exception as e:
                 self.logger.exception(f"Error in {func.__name__}", extra={"input": {"args": args, "kwargs": kwargs}})
                 self._store_sfn_response(
-                    "report", file_name, {"step": func.__name__, "error": str(e), "args": args, "kwargs": kwargs}
+                    "report/errors", file_name, {"step": func.__name__, "error": str(e), "args": args, "kwargs": kwargs}
                 )
                 raise e
 
         return wrapper
 
     def report(self) -> str:
         try:
-            report = dict(errors=[])
-            error_s3_keys = list(
-                self.s3_provider.list_directory(
-                    self.artifact_bucket, self.get_key_prefix(f"schema_migration/{self.execution_id}/report")
+            report = dict(errors=[], migrate_changes=[])
+
+            def retrieve_report_files_from_s3(directory: str):
+                s3_keys = list(
+                    self.s3_provider.list_directory(
+                        self.artifact_bucket,
+                        self.get_key_prefix(f"schema_migration/{self.execution_id}/report/{directory}"),
+                    )
                 )
-            )
-            self.logger.info("Error files found", extra={"error_files": len(error_s3_keys)})
-            for s3_key in error_s3_keys:
-                local_file = os.path.join(self.local_path, "data.json")
-                self.s3_provider.download_file(self.artifact_bucket, s3_key, local_file)
-                with open(local_file, "r") as f:
-                    jn = json.load(f)
-                report["errors"].append(jn)
+                self.logger.info(f"{len(s3_keys)} {directory} files found")
```

Comment:
I think the variable name `directory` is confusing since it's not a full directory and only the suffix.  Maybe `message_type` or something similar.

nit: 
```suggestion
                self.logger.info("Files Found", extra={"directory": directory, "count": len(s3_keys)}")
```